### PR TITLE
Move types to migrations files

### PR DESCRIPTION
**Pull request by Eprince-hub**
Move the `User` type to the migration file to match our recommendation for better linting

### CODE REVIEW

File: app/(auth)/api/login/route.ts [hunk 1]
```diff
@@ -3,11 +3,11 @@ import bcrypt from 'bcrypt';
...
 import { NextRequest, NextResponse } from 'next/server';
 import { createSessionInsecure } from '../../../../database/sessions';
+import { getUserWithPasswordHashInsecure } from '../../../../database/users';
 import {
-  getUserWithPasswordHashInsecure,
   User,
-} from '../../../../database/users';
-import { userSchema } from '../../../../migrations/00006-createTableUsers';
+  userSchema,
+} from '../../../../migrations/00006-createTableUsers';
```
Comment (Eprince-hub) [inline app/(auth)/api/login/route.ts:10]: Import type from the new location

File: app/(auth)/api/register/route.ts [hunk 2]
```diff
@@ -6,9 +6,11 @@ import { createSessionInsecure } from '../../../../database/sessions';
 import {
   createUserInsecure,
   getUserInsecure,
-  User,
 } from '../../../../database/users';
-import { userSchema } from '../../../../migrations/00006-createTableUsers';
+import {
+  User,
+  userSchema,
+} from '../../../../migrations/00006-createTableUsers';
```
Comment (Eprince-hub) [inline app/(auth)/api/register/route.ts:13]: Import type from the new location

File: migrations/00004-createTableAnimalFoods.ts [hunk 3]
```diff
@@ -34,5 +34,5 @@ export async function up(sql: Sql) {
 }
 
 export async function down(sql: Sql) {
-  await sql` DROP TABLE animal_foods `;
+  await sql`DROP TABLE animal_foods`;
```
Comment (Eprince-hub) [inline migrations/00004-createTableAnimalFoods.ts:37]: Trailing empty spaces removed

File: migrations/00006-createTableUsers.ts [hunk 4]
```diff
@@ -1,6 +1,11 @@
 import { Sql } from 'postgres';
 import { z } from 'zod';
 
+export type User = {
+  id: number;
+  username: string;
+};
+
```
Comment (Eprince-hub) [inline migrations/00006-createTableUsers.ts:8]: `User` type is now exported from the migrations file

File: migrations/00008-createTableNotes.ts [hunk 5]
```diff
@@ -1,18 +1,18 @@
 import { Sql } from 'postgres';
 import { z } from 'zod';
 
-export const noteSchema = z.object({
-  title: z.string().min(3).max(100),
-  textContent: z.string().min(5),
-});
-
```
Comment (Eprince-hub) [inline migrations/00008-createTableNotes.ts:8]: Reordering schema and type to follow other files, type first and then schema

**Review by karlhorky — APPROVED** [review]

**Pull request merged**